File: Samples-Java/Samples/src/main/java/fragments/autocomplete/AutoCompleteGettingStartedFragment.java
package fragments.autocomplete;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.Display;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.telerik.android.sdk.R;
import com.telerik.widget.autocomplete.AutoCompleteAdapter;
import com.telerik.widget.autocomplete.CompletionMode;
import com.telerik.widget.autocomplete.CompletionModeContains;
import com.telerik.widget.autocomplete.CompletionModeStartsWith;
import com.telerik.widget.autocomplete.DisplayMode;
import com.telerik.widget.autocomplete.RadAutoCompleteTextView;
import com.telerik.widget.autocomplete.SuggestMode;
import com.telerik.widget.autocomplete.TokenModel;
import java.util.ArrayList;

import activities.ExampleFragment;


public class AutoCompleteGettingStartedFragment extends JsonDataLoadFragment implements ExampleFragment {

    // >> autocomplete-array
    private String[] data = new String[]{"Australia", "Albania","Austria", "Argentina", "Maldives","Bulgaria","Belgium","Cyprus","Italy","Japan",
                                        "Denmark","Finland","France","Germany","Greece","Hungary","Ireland",
                                        "Latvia","Luxembourg","Macedonia","Moldova","Monaco","Netherlands","Norway",
                                        "Poland","Romania","Russia","Sweden","Slovenia","Slovakia","Turkey","Ukraine",
                                        "Vatican City", "Chad", "China", "Chile"};
    // << autocomplete-array

    private RadAutoCompleteTextView autocomplete;
    private AutoCompleteAdapter adapter;

    @Override
    public String title() {
        return "Getting Started";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View rootView = inflater.inflate(R.layout.autocomplete_getting_started, container, false);
        // >> autocomplete-load
        autocomplete = (RadAutoCompleteTextView) rootView.findViewById(R.id.autocmp);
        // << autocomplete-load

        // >> autocomplete-suggest-display
        autocomplete.setSuggestMode(SuggestMode.SUGGEST);
        autocomplete.setDisplayMode(DisplayMode.PLAIN);
        // << autocomplete-suggest-display

        // >> autocomplete-adapter
        adapter = new AutoCompleteAdapter(this.getContext(),this.getTokenModelObjects(), R.layout.suggestion_item_layout);
        adapter.setCompletionMode(new CompletionModeStartsWith());
        autocomplete.setAdapter(adapter);
        // << autocomplete-adapter

        Display display = this.getActivity().getWindowManager().getDefaultDisplay();
        int height  =  display.getHeight();
        autocomplete.setSuggestionViewHeight(height/4);

        this.setButtonAction(rootView);

        return rootView;
    }

    private void setButtonAction(View rootView){
        Button btnSuggest = (Button)rootView.findViewById(R.id.suggestButton);
        btnSuggest.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setSuggestMode(SuggestMode.SUGGEST);
            }
        });

        Button btnSuggestAppend = (Button)rootView.findViewById(R.id.suggestAppendButton);
        btnSuggestAppend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setSuggestMode(SuggestMode.SUGGEST_APPEND);
            }
        });

        Button btnAppend = (Button)rootView.findViewById(R.id.appendButton);
        btnAppend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setSuggestMode(SuggestMode.APPEND);
            }
        });

        Button btnStartsWith = (Button)rootView.findViewById(R.id.startsWithButton);
        btnStartsWith.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                adapter.setCompletionMode(new CompletionModeStartsWith());
                autocomplete.resetAutocomplete();
            }
        });
        Button btnContains = (Button)rootView.findViewById(R.id.containsButton);
        btnContains.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                adapter.setCompletionMode(new CompletionModeContains());
                autocomplete.setSuggestMode(SuggestMode.SUGGEST);
                autocomplete.resetAutocomplete();
            }
        });
        Button btnTokens = (Button)rootView.findViewById(R.id.tokens_mode_btn);
        btnTokens.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setDisplayMode(DisplayMode.TOKENS);
                autocomplete.resetAutocomplete();
            }
        });

        Button btnPlain = (Button)rootView.findViewById(R.id.plain_mode_btn);
        btnPlain.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                autocomplete.setDisplayMode(DisplayMode.PLAIN);
                autocomplete.resetAutocomplete();
            }
        });

    }

    // >> autocomplete-token-models
    private ArrayList<TokenModel> getTokenModelObjects() {
        ArrayList<TokenModel> feedData = new ArrayList<TokenModel>();
        for(int i = 0; i < this.data.length; i++){
            TokenModel token = new TokenModel(this.data[i], null);
            feedData.add(token);
        }

        return feedData;
    }
    // << autocomplete-token-models
}

File: Samples-Xamarin/Fragments/Autocomplete/AutoCompleteGettingStartedFragment.cs
using System;
using System.Collections.Generic;
using Android.Views;
using Android.Widget;
using Com.Telerik.Widget.Autocomplete;


namespace Samples
{
	public class AutoCompleteGettingStartedFragment : Android.Support.V4.App.Fragment, ExampleFragment
	{
		private RadAutoCompleteTextView autocomplete;
		private AutoCompleteAdapter adapter;
		private List<String> data;

		public AutoCompleteGettingStartedFragment() : base()
		{
			// >> autocomplete-array-xamarin
			this.data = new List<string>() { "Australia", "Albania","Austria", "Argentina", "Maldives","Bulgaria","Belgium","Cyprus","Italy","Japan",
										"Denmark","Finland","France","Germany","Greece","Hungary","Ireland",
										"Latvia","Luxembourg","Macedonia","Moldova","Monaco","Netherlands","Norway",
										"Poland","Romania","Russia","Sweden","Slovenia","Slovakia","Turkey","Ukraine",
										"Vatican City", "Chad", "China", "Chile" };
			// << autocomplete-array-xamarin
		}

		public string Title()
		{
			return "Getting Started";
		}

		public override Android.Views.View OnCreateView(Android.Views.LayoutInflater inflater, Android.Views.ViewGroup container, Android.OS.Bundle savedInstanceState)
		{
			View rootView = inflater.Inflate(Resource.Layout.fragment_autocomplete_getting_started,container,false);

			// >> autocomplete-load-xamarin
			this.autocomplete = (RadAutoCompleteTextView)rootView.FindViewById(Resource.Id.autocmp);
			// << autocomplete-load-xamarin

			// >> autocomplete-suggest-display-xamarin
			this.autocomplete.SuggestMode = SuggestMode.Suggest;
			this.autocomplete.DisplayMode = DisplayMode.Plain;
			// << autocomplete-suggest-display-xamarin

			// >> autocomplete-adapter-xamarin
			this.adapter = new AutoCompleteAdapter(this.Context, this.GetTokenObjects(), Java.Lang.Integer.ValueOf(Resource.Layout.suggestion_item_layout));
			this.adapter.CompletionMode = CompletionMode.StartsWith;
			this.autocomplete.Adapter = this.adapter;
			// << autocomplete-adapter-xamarin

			Display display = this.Activity.WindowManager.DefaultDisplay;
			int height = display.Height;
			this.autocomplete.SuggestionViewHeight = height / 4;

			this.SetButtonActions(rootView);

			return rootView;
		}

		// >> autocomplete-token-models-xamarin
		private List<TokenModel> GetTokenObjects()
		{
			List<TokenModel> feedData = new List<TokenModel>();
			for (int i = 0; i < this.data.Count; i++)
			{
				TokenModel token = new TokenModel(this.data[i], null);
				feedData.Add(token);
			}

			return feedData;
		}
		// << autocomplete-token-models-xamarin

		private void SetButtonActions(View rootView)
		{
			Button btnSuggest = (Button)rootView.FindViewById(Resource.Id.suggestButton);
			btnSuggest.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.SuggestMode = SuggestMode.Suggest;
			};

			Button btnSuggestAppend = (Button)rootView.FindViewById(Resource.Id.suggestAppendButton);
			btnSuggestAppend.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.SuggestMode = SuggestMode.SuggestAppend;
			};

			Button btnAppend = (Button)rootView.FindViewById(Resource.Id.appendButton);
			btnAppend.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.SuggestMode = SuggestMode.Append;
			};

			Button btnStartsWith = (Button)rootView.FindViewById(Resource.Id.startsWithButton);
			btnStartsWith.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.Adapter.CompletionMode = CompletionMode.StartsWith;
			};

			Button btnContains = (Button)rootView.FindViewById(Resource.Id.containsButton);
			btnContains.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.Adapter.CompletionMode = CompletionMode.Contains;
			};

			Button btnTokens = (Button)rootView.FindViewById(Resource.Id.tokensButton);
			btnTokens.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.DisplayMode = DisplayMode.Tokens;
			};

			Button btnPlain = (Button)rootView.FindViewById(Resource.Id.plainButton);
			btnPlain.Click += (object sender, EventArgs e) =>
			{
				this.autocomplete.DisplayMode = DisplayMode.Plain;
			};

		}
	}
}
